### PR TITLE
Resolves #197 tag component keyboard focusable remove button

### LINKED ISSUE #197: The Tag component remove button needs to be keyboard accessible
The `<Tag/>` component remove button is not currently keyboard accessible.

## Requirements

* The remove button should be focusable by keyboard users
* The remove button should be interactive with the enter key and spacebar

## Potential Fixes

* Update the HTML to use the `<button>` element instead of the `<a>`
* __or__ use the existing `<a>` element and add `role="button"` alongside a relevant value for `href` - `href="javascript:void(0):"` should do the trick

### PR DESCRIPTION
**Pull request by nicklemmon**
Resolves #197

## What Changed

* Added new `handleKeydown` method to enable spacebar handling
* Added `role="button"` and an arbitrary value to the `to` prop to allow keyboard focusing
* Added supporting test for 

## How to Test

* Go to http://localhost:9001/?selectedKind=Feedback%7CTag&selectedStory=with%20remove%20action&full=0&addons=1&stories=1&panelRight=0&addonPanel=storybook%2Factions%2Factions-panel
* Click immediately before the tag component instance, hit the tab key and the button should have focus
* Hit the spacebar or enter key when focusing on the remove button and the action logger should show the mock function

### CODE REVIEW

File: packages/matchbox/src/components/Tag/Tag.js [hunk 1]
```diff
@@ -43,7 +47,10 @@ class Tag extends Component {
     const closeMarkup = onRemove
       ? <UnstyledLink
         className={styles.Close}
-        onClick={onRemove}>
+        onClick={onRemove}
+        onKeyDown={(e) => this.handleKeydown(e, onRemove)}
```
Comment (jonambas) [inline packages/matchbox/src/components/Tag/Tag.js]: We have keydown helpers in `helpers/keyEvents`. Could slim some of this down.

```js
onKeyDown={onKey('space', props.onRemove)}
```

Comment (nicklemmon) [reply packages/matchbox/src/components/Tag/Tag.js]: Ooh cool. I’ll definitely check those out and clean it up.

Comment (nicklemmon) [reply packages/matchbox/src/components/Tag/Tag.js]: Alrighty, swapped it out. Nice API there!

**Comment by nicklemmon** [comment]
@jonambas for some reason, the Jest test to check for the spacebar firing the method passed in to `onRemove` doesn't work with the helper functions! But it does work otherwise. If you have any ideas I'm all ears but trying to figure that out now.

**Comment by nicklemmon** [comment]
OK, got it figured out - Enzyme's simulate method does not account for `shiftKey` by default and thus returns that value as `undefined` rather than `false` by default. Kind of weird!

**Comment by nicklemmon** [comment]
Nice, thanks.

So it looks like the browser default focus styles are present, which is totally kosher. We could chat about adding a custom focus style. One that I've used in the past (this is global CSS and isn't scoped, so there's that):

```
*:focus {
  outline: 1px dashed currentColor;
}
```

That ends up handling about 90% of cases. Each component might require some custom tweaking in certain scenarios, but not too bad.

For another PR, though!

**Pull request merged**